File: CS/ASP.NET/MemberShipRoleProvider/MemberShipRoleProvider/MasterPage.master.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

public partial class MasterPage : System.Web.UI.MasterPage
{
    protected void Page_Load(object sender, EventArgs e)
    {
        if (!IsPostBack)
        {
            LoginStatus1.Visible = false;
            if (Roles.IsUserInRole("Admin") || Roles.IsUserInRole("User"))
            {
                login.Visible = false;
                LoginStatus1.Visible = true;
            }
            if (this.Context.User.Identity.Name != null)
            {
                login.Visible = false; LoginStatus1.Visible = true;
            }
            if (Roles.IsUserInRole("Admin"))
            {
                Menu1.Items[1].Text = "Admin";
            }
            else
            {
                Menu1.Items[1].Text = "";
            }
        }    
    }
}

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Project/VB/Online Travel Agency/Online Travel Agency/MasterPage.master.vb
Public Class MasterPage
    Inherits System.Web.UI.MasterPage

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        If (LoginName1.Page.User.Identity.IsAuthenticated) Then
            hlChangePassowrd.Visible = True
            LoginStatus1.Visible = True
            Login1.Visible = False
        Else
            hlChangePassowrd.Visible = False
            LoginStatus1.Visible = False
            Login1.Visible = True
        End If
    End Sub

    Protected Sub LoginButton_Click(ByVal sender As Object, ByVal e As EventArgs)

    End Sub
End Class